test/recipes/02-test_errstr.t

```diff
@@ -132,3 +73,58 @@ sub split_error {
 
     return @erritems;
 }
+
+# Compares the first argument as string to each of the arguments 3 and on,
+# and returns an array of two elements:
+# 0:  True if the first argument matched any of the others, otherwise false
+# 1:  A string describing the test
+# The returned array can be used as the arguments to Test::More::ok()
+sub match_any {
+    my $first = shift;
+    my $desc = shift;
+    my @strings = @_;
+
+    if (scalar @strings > 1) {
+        $desc = "match '$first' ($desc) with one of ( '"
+            . join("', '", @strings) . "' )";
+    } else {
+        $desc = "match '$first' ($desc) with '$strings[0]'";
+    }
+
+    return ( scalar( grep { $first eq $_ } @strings ) > 0,
+             $desc );
+}
+
+sub match_opensslerr_reason {
+    my $errnum = shift;
+    my @strings = @_;
```

Comment:
This variable actually obtains `$errnum` (which is likely not intended, see comment below at the end of `match_syserr_reason`) and the two strings computed in the below function.